### PR TITLE
#2672 ragdoll/dead chams FIXED

### PR DESCRIPTION
**Pull request by wirekuro**
unknown

### CODE REVIEW

**Comment by wemanzoz** [comment]
ok this is cool, @danielkrupinski possible merge?

**Comment by 3Ld4D** [comment]
Why are you setting it to 100. 2 is enough for ragdoll.

**Comment by wirekuro** [comment]
> Why are you setting it to 100. 2 is enough for ragdoll.

At the setting 2 it was buggy for me, I randomly set it to 100 and its good.

File: Osiris/Config.cpp [hunk 1]
```diff
@@ -888,6 +889,7 @@ static void to_json(json& j, const Config::Visuals& o)
 
     WRITE("Disable post-processing", disablePostProcessing);
     WRITE("Inverse ragdoll gravity", inverseRagdollGravity);
+    WRITE("rdrawothermodels", rdrawothermodels);
```
Comment (mkot2) [inline Osiris/Config.cpp:892]: why bother with all of this, you cant change this anyway so just delete it from config altogether and always set drawothermodels to 100

File: Osiris/Hacks/Visuals.cpp [hunk 2]
```diff
@@ -129,6 +129,12 @@ void Visuals::playerModel(FrameStage stage) noexcept
     }
 }
 
+void Visuals::rdrawothermodels() noexcept
+{
+    static auto rdrawothermodels = interfaces->cvar->findVar("r_drawothermodels");
+    rdrawothermodels->setValue(config->visuals.rdrawothermodels ? 100 : 0);
```
Comment (mkot2) [inline Osiris/Hacks/Visuals.cpp:135]: why make a function for this, can't you do this directly in hooks?

Comment (wemanzoz) [reply Osiris/Hacks/Visuals.cpp:135]: zzzz why dont you do it yourself then, stop complaining on others work

Comment (wirekuro) [reply Osiris/Hacks/Visuals.cpp:135]: @mkot2 bro what's the point its working lol

Comment (mkot2) [reply Osiris/Hacks/Visuals.cpp:135]: > 
> 
> zzzz why dont you do it yourself then, stop complaining on others work

how am i complaining bro i'm just sayin

**Comment by debooga** [comment]
this is not safe at all and can get you untrusted banned

**Comment by wirekuro** [comment]
> this is not safe at all and can get you untrusted banned

there is vac bypass + untrusted bans are cuz bad injectors

**Comment by 3Ld4D** [comment]
> > this is not safe at all and can get you untrusted banned
> 
> there is vac bypass + untrusted bans are cuz bad injectors

r_drawothermodels is checked you need to spoof it.
https://www.unknowncheats.me/forum/counterstrike-global-offensive/173295-convars-checked-valve-servers.html

**Comment by degeneratehyperbola** [comment]
That breaks AUG/SG553 crosshair

**Comment by debooga** [comment]
please dont use this pr cuz it can cause untrusted bans

**Comment by degeneratehyperbola** [comment]
I actually hooked `r_drawothermodels` properly, but without a constant known return address this wouldn't make much sense
Btw, do you know any other methods to know where `r_drawothermodels->getInt()` is called from?

**Comment by debooga** [comment]
> I actually hooked `r_drawothermodels` properly, but without a constant known return address this wouldn't make much sense
> Btw, do you know any other methods to know where `r_drawothermodels->getInt()` is called from?
> I really don't want to rewrite the entirety of chams system for endScene or drawModel

"You need to hook r_drawmodelstatsoverlay and return true when _ReturnAddress == 85 C0 75 54 8B 0D ? ? ? ?." #2672 0suddenly0

**Comment by 3Ld4D** [comment]
> I actually hooked `r_drawothermodels` properly, but without a constant known return address this wouldn't make much sense
> Btw, do you know any other methods to know where `r_drawothermodels->getInt()` is called from?
> I really don't want to rewrite the entirety of chams system for endScene or drawModel

EndScene is ass, and DrawModel is called by DrawModelExecute so u still need to fix it.

**Comment by degeneratehyperbola** [comment]
@3Ld4D Yep, you're right
@debooga As i have said the return address isn't constant, but nevertheless I tried and haven't had much success.

**Comment by cristeigabriel** [comment]
More ideally take care of either r_drawmodelstatsoverlay or mat_wireframe for convention sake, or at this point just handle renderable models yourself. The aforementioned are enough of a reference for one to make sense of what I'm saying.

**Pull request closed without merging**